### PR TITLE
Logging setup

### PR DESCRIPTION
**Pull request by blowmage**
Create core Logging entities such as Project, Connection, Credentials, Resource, Sink, and Metric. Most resources and API calls are accounted for, except for Entries.

### CODE REVIEW

File: lib/gcloud.rb [hunk 1]
```diff
@@ -313,4 +313,39 @@ def search scope: nil
...
+  #   * `https://www.googleapis.com/auth/logging.admin`
+  #
+  # @return [Gcloud::Logging::Project]
+  #
+  # @example
+  #   require "gcloud"
+  #
+  #   gcloud = Gcloud.new
+  #   logging = gcloud.logging
+  #   topic = logging.topic "my-topic"
```
Comment (quartzmo) [inline lib/gcloud.rb]: Change from this pubsub example to `logging.resources`, etc.

Comment (blowmage) [inline lib/gcloud.rb]: Ugh, I thought I caught all of those. Updated and removed. These code examples will need to be revisited once we get more of the logging service implemented.

File: lib/gcloud/logging/project.rb [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+      # @return [Gcloud::Logging::Sink]
+      #
+      # @example
+      #   require "gcloud"
+      #
+      #   gcloud = Gcloud.new
+      #   logging = gcloud.logging
+      #   sink = logging.create_sink "my-sink"
+      #
+      def create_sink name, destination: nil, filter: nil, version: :unspecified
```
Comment (quartzmo) [inline lib/gcloud/logging/project.rb]: The API docs state: "An unspecified version format will default to V2." If this is the case, do we need/want to include `unspecified` as an option?

File: lib/gcloud/logging/project.rb [hunk 3]
```diff
@@ -0,0 +1,362 @@
...
+        version = Sink::VERSIONS[version] if Sink::VERSIONS[version]
```
Comment (quartzmo) [inline lib/gcloud/logging/project.rb]: In other services we have done this type of conversion differently, in protected methods in Connection. I'm good with either way, just pointing this out.

File: lib/gcloud/logging/project.rb (hunk 2 above)
Comment (blowmage) [inline lib/gcloud/logging/project.rb]: It seems like the default behavior is when unspecified it defaults to V2. That could change over time, so the way I read it is that the default value is `unspecified` which will then become `V2`. So I thought it made sense to make the default `unspecified`. But right now I'm just trying to establish the service API defaults, we can always refine the ruby code defaults as we polish the ruby API. Getting some acceptance tests would help clarify this I bet.

File: lib/gcloud/logging/resource/list.rb [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+      ##
+      # Resource::List is a special case Array with additional values.
+      class List < DelegateClass(::Array)
+        ##
+        # If not empty, indicates that there are more records that match
+        # the request and this value should be passed to continue.
+        attr_accessor :token
+
+        ##
+        # Create a new Zone::List with an array of Zone instances.
```
Comment (quartzmo) [inline lib/gcloud/logging/resource/list.rb]: All three List docs have residual Zone/zone mentions.

File: lib/gcloud/logging/project.rb (hunk 3 above)
Comment (blowmage) [inline lib/gcloud/logging/project.rb]: Yeah, we can make those changes in a future refactor and not change the existing ruby API. Sometimes I wait until I see something duplicated 3 or more times before refactoring.

File: lib/gcloud/logging/resource.rb [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+      # documentation.
+      def description
+        @gapi["description"]
+      end
+
+      ##
+      # A set of labels that can be used to describe instances of this monitored
+      # resource type.
+      def labels
+        # TODO: Make a proper Label class to represent this structure...
```
Comment (quartzmo) [inline lib/gcloud/logging/resource.rb]: Yes, curious to see how we can represent the somewhat awkward `LabelDescriptor`...

Comment (blowmage) [inline lib/gcloud/logging/resource.rb]: Yeah, not looking forward to that.. :)

**Comment by quartzmo** [comment]
What a strong start to Logging. I know you put a lot of work into this! Great progress.

**Comment by blowmage** [comment]
Thanks for being so diligent in the code review @quartzmo. I'll try to keep future PRs smaller and easier to review.

**Pull request merged**